pkg/tar/tar.go

```diff
@@ -38,10 +39,47 @@ var ErrNotSupportedPlatform = errors.New("platform and architecture is not suppo
 // root of the tar file and should be cleaned (for example using filepath.Clean)
 type PathWhitelistMap map[string]struct{}
 
-// extractTar extracts a tarball (from a tar.Reader) into the root directory
-// if pwl is not nil, only the paths in the map are extracted.
-// If overwrite is true, existing files will be overwritten.
-func extractTar(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
+type FilePermissionsEditor func(string, int, int, byte, os.FileInfo) error
+
+func NewUidShiftingFilePermEditor(uidRange *uid.UidRange) (FilePermissionsEditor, error) {
+	user, err := user.Current()
```

Comment:
This seems to fail on [Semaphore](https://semaphoreci.com/coreos/rkt/branches/pull-request-1616/builds/2):
```
		run: error setting up stage0: error preparing stage1: 
                    error rendering tree image: treestore: cannot render aci:
                    error extracting ACI: extracttar error: exit status 1, output:
                    error determining current user:
                    user: lookup userid 0: no such file or directory
```

However, I cannot reproduce it while ssh'ing on Semaphore...